cmd/kured/main.go

```diff
@@ -531,11 +538,21 @@ func rebootAsRequired(nodeID string, rebootCommand []string, sentinelCommand []s
 			continue
 		}
 
-		drain(client, node)
-		invokeReboot(nodeID, rebootCommand)
-		for {
-			log.Infof("Waiting for reboot")
-			time.Sleep(time.Minute)
+		isDrained = drain(client, node)
+		if isDrained {
+			invokeReboot(nodeID, rebootCommand)
+			for {
+				log.Infof("Waiting for reboot")
+				time.Sleep(time.Minute)
+			}
+		} else {
+			log.Infof("Uncordon %s", node.GetName())
+			uncordon(client, node)
+			deleteFlag := newCommand("/usr/bin/nsenter", "-m/proc/1/ns/mnt", "/bin/rm", rebootSentinelFile)
```

Comment:
I think it would be best to use the available functions for wrapping things properly (so that it works on all OSes).

Next, what about the cases where the sentinel is a command instead of a sentinel file?